app/services/proofing/aamva/proofer.rb

```diff
@@ -73,30 +87,30 @@ def parse_verification_errors(verification_response)
         errors
       end
 
-      def verified_attributes(verification_response)
-        attributes = Set.new
-        results = verification_response.verification_results
-
-        attributes.add :address if address_verified?(results)
+      def requested_attributes(verification_response)
+        attributes = verification_response.
+          verification_results.filter { |_, verified| !verified.nil? }.
```

Comment:
Is this intended to reflect the requested attributes based on what we see from the response object? The response object iterates over all possible attributes so it will include attributes even if we did not request them:

https://github.com/18F/identity-idp/blob/5cb708a9d2d95030bfa835f9d91f5483cb110f75/app/services/proofing/aamva/response/verification_response.rb#L96-L105

Reply:
Yes, that's what the `filter` is about -- `verification_results` will have all attributes present, but any missing attributes will have `nil` values

Reply:
That _should_ work for this issue. 2 things to keep in mind:

1. When AAMVA fails to match a DLN it returns a false match indicator for DLN and no match indicator for other attributes. In that case it will appear that we only requested DLN even when we sent a request for all of the PII including issue and expiration date
2. Cloudwatch has a tough time handling array parameters and parsing them into something useful. If we could find a way to return keys/values here that may be easier to query when we are trying to visualize this. The `verified_attributes` value is an array so we can use it in get-to-yes IIRC

Reply:
Ooh, I did not know that about DLN. It would be a little more work to look at the request itself (currently the proofer just does `client.send_verification_request`, and doesn't actually get a copy of the request data).

I can change `requested_attributes` to a hash no prob, I might poke around and see if it is worth doing `verified_attributes` as well